### PR TITLE
Clone domain tweaks

### PR DESCRIPTION
**Pull request by snopoke**
@czue

### CODE REVIEW

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 1]
```diff
@@ -187,11 +187,23 @@ def copy_ucr_datasources(self, new_domain, old_domain):
 
     def save_couch_copy(self, doc, new_domain=None):
         old_id = doc._id
-        del doc._id
+        doc._id = doc.get_db().server.next_uuid()
```
Comment (snopoke) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: Had issues when actually running it on the server where the doc ID wasn't set during save. Wasn't able to reproduce locally.

Comment (czue) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: no biggie but is this better than `uuid.uuid4().hex`? i really don't like the cloudant uuids that look very similar.

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 2]
```diff
@@ -187,11 +187,23 @@ def copy_ucr_datasources(self, new_domain, old_domain):
...
+        attachments = {}
+        if "_attachments" in doc and doc['_attachments']:
+            attachemnt_stubs = doc["_attachments"]
+            del doc['_attachments']
+            attachments = {k: doc.get_db().fetch_attachment(old_id, k) for k in attachemnt_stubs}
+
         doc.save()
+
+        for k, attach in attachments.items():
+            doc.put_attachment(attach, name=k, content_type=attachemnt_stubs[k]["content_type"])
```
Comment (czue) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: is there anything we need to do with blob fields? (maybe not relevant for any data types we're moving over... yet)

**Comment by czue** [comment]
:+1: on :white_check_mark:

File: corehq/apps/hqadmin/management/commands/clone_domain.py (hunk 2 above)
Comment (snopoke) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: Good question. Not applicable at this stage. The only models we're copying
that could have attachments are `Domain` and `Application`. Application's
are copied using the `import_app` function which already handles
attachments (and which is being updated to handle blobs). Domains is likely
going to be near the end the of blob db migration list.

**Pull request merged**